### DISCUSSION TITLE
How to prevent activating multiple license file in offline environment?

### DISCUSSION THREAD

**Discussion by w4-donguk** [keygen-sh/community#88] [Q&A] [answered]
Hello, I'd like to apply the offline license from on limited number of machine.

I read the offline license section on doc, and I understood that we need to check out the license file and pass it to customer to activate the license.

but I'm curious about this: can it prevent activating from multiple machines with a single license file?
For example, if they copy the license file and activate on multiple machines, Will they all activate?

I'd like to limit the number of machine, Do any keygen feature support this?
It seems like license file are not including any machine-specific information(like mac-address), so I wonder if this is possible, any comment would be appreciated.

Thanks

**Comment by ezekg** [comment, marked as answer]
You can use [a machine file](https://keygen.sh/docs/api/machines/#machines-check-out) instead, which is locked to a specific license via its key AND to a specific machine via its fingerprint. Meaning the machine file can't be shared, since fingerprints will differ from machine to machine, causing the decryption to fail on all but the actual machine. You can include the license inside of the machine file during checkout using the `?include=license` query parameter.
